src/trajectory_data_handling/trajectory_sql.cpp

```diff
@@ -57,22 +59,67 @@ namespace trajectory_data_handling {
             case query_purpose::load_simplified_trajectory_information_into_datastructure:
                 table_name = "simplified_trajectory_information";
                 break;
+
             default:
                 std::cout << "Error in switch statement in load_database_into_datastructure." << '\n';
         }
 
-        query << "SELECT trajectory_id, timestamp, longitude, latitude FROM " << table_name;
+        if(id.empty()) {
+            query << "SELECT trajectory_id, timestamp, longitude, latitude FROM " << table_name;
+        }
+        else {
+            query << "SELECT trajectory_id, timestamp, longitude, latitude FROM " << table_name << " WHERE trajectory_id IN (";
+            for(auto i = id.begin(); i != id.end(); ++i) {
+                query << *i;
+                if (std::next(i) != id.end()) {
+                    query << ",";
+                }
+            }
+            query << ")";
+        }
         trajectory_data_handling::query_handler::run_sql(query.str().c_str(), purpose);
     }
 
+    void trajectory_manager::spatial_query_on_rtree_table(query_purpose purpose, std::tuple<float, float> longitudeRange, std::tuple<float, float> latitudeRange, std::tuple<float, float> timestampRange) {
+        std::string table_name{};
+        std::stringstream query{};
+
+        switch(purpose) {
+            case query_purpose::load_original_rtree_into_datastructure:
+                table_name = "trajectory_rtree";
+                break;
+            case query_purpose::load_simplified_rtree_into_datastructure:
+                table_name = "simplified_trajectory_rtree";
+                break;
+
+            default:
+                std::cout << "Error in switch statement in spatial_query_on_rtree_table." << '\n';
+        }
+
+        query << "SELECT id FROM " << table_name << " WHERE minLongitude>=" << get<0>(longitudeRange) << " AND maxLongitude<=" << get<1>(longitudeRange) << " AND minLatitude>=" << get<0>(latitudeRange) << " AND maxLatitude<=" << get<1>(latitudeRange) << " AND minTimestamp>=" << get<0>(timestampRange) << " AND maxTimestamp<=" << get<1>(timestampRange) << ";";
+        trajectory_data_handling::query_handler::run_sql(query.str().c_str(), purpose);
+
+        if (trajectory_data_handling::query_handler::id.empty()){}
```

Comment:
Hvorfor ikke !empty og så kill den else branch?
```suggestion
        if (!trajectory_data_handling::query_handler::id.empty()){}
```

Reply:
lavet om